ledger/acctupdates.go

```diff
@@ -1794,7 +1814,10 @@ func (au *accountUpdates) lookupWithRewards(rnd basics.Round, addr basics.Addres
 			// Check if this is the most recent round, in which case, we can
 			// use a cache of the most recent account state.
 			if offset == uint64(len(au.deltas)) {
-				return macct.data, nil
+				if full && macct.data.ExtendedAssetHolding.Count != 0 {
+					macct.data.Assets, macct.data.ExtendedAssetHolding, err = au.accountsq.loadHoldings(macct.data.ExtendedAssetHolding)
```

Comment:
in order to make sure that the return is valid, you need to join it with the round number and compare it with `currentDbRound`. Otherwise, you might have just created a mixed-up balance record of old account data with new extended holding.

Reply:
done